### ISSUE TITLE
Abstract out Google Auth

### ISSUE THREAD

**Issue by beaulebens** [beaulebens/keyring#27]
We now have Google Analytics, Google Contacts, and YouTube Services, all of which rely basically on the same central (Google) Auth system. Should probably abstract/standardize things, so that each of these inherit from one central service which handles all the duplicate stuff.

A quick comparison indicates some similarities across:

- `SCOPE` and `ACCESS_TYPE` handling and filtering.
- Modified UI (with extended instructions because Google's Console is super complicated).
- All auth/token endpoints.
- Refresh tokens.
- Token meta.
- Connection testing.

So basically they are the same, minus some labeling and specific scopes.
